### PR TITLE
Add BWC tests for running against distribution bundle. 

### PR DESCRIPTION
**Pull request by zelinh**
### Description
Add option to run BWC tests in distribution level, which is running tests in the test cluster with latest distribution bundle installed and all plugins (included in the latest manifest) present. 

The command to run BWC tests in distribution level would be `./gradlew bwcTestSuite -Dtests.security.manager=false -PcustomDistributionDownloadType=bundle`
If property `customDistributionDownloadType` is not set nor set to `bundle`, the BWC tests would be default to run in plugins level as previously configured. 

This won't affect current plugin level tests written in the `./scripts/bwctest.sh`.

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-build/issues/2870
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: integ-test/build.gradle [hunk 1]
```diff
@@ -255,27 +259,55 @@ String bwcRemoteFile = "https://ci.opensearch.org/ci/dbc/bundle-build/1.1.0/2021
...
-                        File getAsFile() {
-                            File dir = new File('./integ-test/' + bwcFilePath + bwcVersion)
-                            if (!dir.exists()) {
-                                dir.mkdirs()
-                            }
-                            File f = new File(dir, bwcSqlPlugin)
-                            if (!f.exists()) {
-                                new URL(bwcRemoteFile).withInputStream{ ins -> f.withOutputStream{ it << ins }}
+            if (bwcBundleTest) {
+                versions = ["1.3.2", "2.5.0"]
```
[the remark this answers is not in the document]
Comment (zelinh) [reply integ-test/build.gradle:264]: @penghuo 
1. Yes. The only difference would be one is 3 digits and one is the `bwcVersion` is four digits.
2. For this one, `opensearch_version` contains `-SNAPSHOT` which the framework will use to download the bundle through CI URL. However, our current URL for CI bundle are only using non-snapshot version for example, `https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/1.3.2/latest/linux/x64/tar/dist/opensearch/opensearch-1.3.2-linux-x64.tar.gz`

[the remark this answers is not in the document]
Comment (penghuo) [reply integ-test/build.gradle:264]: > For this one, opensearch_version contains -SNAPSHOT which the framework will use to download the bundle through CI URL. However, our current URL for CI bundle are only using non-snapshot version for example, https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.5.0/latest/linux/x64/tar/dist/opensearch/opensearch-2.5.0-linux-x64.tar.gz

could we remove -SNAPSHOT from opensearch_version? my concern is if opensearch_version bump to 2.6.0-SNAPSHOT, we need to change 2.5.0 to 2.6.0 in bwc, right?

[the remark this answers is not in the document]
Comment (zelinh) [reply integ-test/build.gradle]: Yes, that would make sense. Initially I was thinking include this as part of the auto increment version PR. Now I updated with the dynamic version.

**Review by penghuo — APPROVED** [review]
Thanks for the change!

**Review by dai-chen — APPROVED** [review]

**Pull request merged**